packages/flutter/lib/src/material/switch.dart

```diff
@@ -213,31 +213,24 @@ class Switch extends StatefulWidget {
 }
 
 class _SwitchState extends State<Switch> with TickerProviderStateMixin {
-  Map<LocalKey, ActionFactory> _actionMap;
+  Map<Type, Action<Intent>> _actionMap;
 
   @override
   void initState() {
     super.initState();
-    _actionMap = <LocalKey, ActionFactory>{
-      ActivateAction.key: _createAction,
+    _actionMap = <Type, Action<Intent>>{
+      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
     };
   }
 
-  void _actionHandler(FocusNode node, Intent intent){
+  void _actionHandler(ActivateIntent intent) {
     if (widget.onChanged != null) {
       widget.onChanged(!widget.value);
     }
-    final RenderObject renderObject = node.context.findRenderObject();
+    final RenderObject renderObject = primaryFocus.context.findRenderObject();
```

Comment:
(This comment also applies to some of the other widgets that have been modified in a similar fashion)

Reply:
Yeah, I guess it's not guaranteed.  It's highly likely, since the action dispatcher is going to usually be calling it from an intent invoked from the primary focus.

I modified this and others to use the local context in the state.